app/bundles/EmailBundle/Helper/MailHelper.php

```diff
@@ -927,22 +927,30 @@ public function addTo($address, $name = null): bool
     /**
      * Set CC address(es).
      *
-     * @param array<string,?string> $addresses
-     * @param ?string               $name
-     *
-     * //TODO: there is a bug here, the name is not passed in CC nor in the array of addresses, we do not handle names for CC
+     * @param array<string|int,?string> $addresses Array of emails as values or keys
+     * @param ?string                   $name      Default name for addresses without specified names
      */
     public function setCc($addresses, $name = null): bool
     {
         $this->checkBatchMaxRecipients(count($addresses), 'cc');
 
         try {
             $ccAddresses = [];
-            // The email addresses are stored in the array keys not the values
-            // The name of the CC is passed in the function and not in the array
-            foreach ($addresses as $address => $noName) {
-                $ccAddresses[] = (new AddressDTO($address, $name))->toMailerAddress();
+
+            foreach ($addresses as $key => $value) {
+                // Check if we have an indexed array (numeric keys)
+                if (is_numeric($key)) {
+                    $address     = $value;
+                    $addressName = $name;
+                } else {
+                    // We have an associative array (email => name)
+                    $address     = $key;
+                    $addressName = $value ?: $name; // Use provided name or default
+                }
```

Comment:
Can we create a function to reduce the redundancy? The same code can be found here https://github.com/mautic/mautic/pull/14719/files#diff-d1949283f925ccb79831f1331dca03d32cadc4cef0e05f0e9bed3f55d5d410aeR1000-R1007

Reply:
Done! Thanks for your review